spec/presenters/two_factor_auth_code/phoneable.spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+phone_update_path = 'a/link/to/update'
+
+describe TwoFactorAuthCode::Phoneable do
+  let(:presenter) do
+    TwoFactorAuthCode::PhoneDeliveryPresenter.new(attributes_for(:generic_otp_presenter))
+  end
+
+  let(:unconfirmed) do
+    data = attributes_for(:generic_otp_presenter).merge!(
+      reenter_phone_number_path: phone_update_path,
+      unconfirmed_phone: true
+    )
+
+    TwoFactorAuthCode::PhoneDeliveryPresenter.new(data)
+  end
+
+  describe '#phone_fallback_link' do
+    it 'returns a link for voice delivery if user initially selected sms' do
+      output = presenter.phone_fallback_link
+
+      expect(output).to include(t('links.two_factor_authentication.voice'))
+      expect(output).to match(/<a href/)
+    end
+
+    it 'return a link for sms delivery if user initially selected voice' do
+      output = presenter.phone_fallback_link
+
+      expect(output).to include(t('links.two_factor_authentication.sms'))
+    end
```

Comment:
These 2 specs are not specific enough. They are both passing, but the second one should fail because the delivery method is still `sms` for this second test. The presenter was not updated to set the `delivery_method` to `voice` for this test. The test is passing because it is only testing one part of the output, the link, which in this case happens to share text in the first part of the output. When the delivery method is `sms`, the output starts by saying `If you can't get text messages`, and the test is trying to make sure that the link text says `text message`, but it's finding the match in the first part of the sentence and passing.

At a minimum, we want to test the first part of the sentence, which is different in both cases:
```ruby
expect(output).to include t('instructions.2fa.voice.fallback_html')
```

For the full experience, I would test the full link:
```ruby
it 'returns a link for voice delivery if user initially selected sms' do
  output = presenter.phone_fallback_link

  link = "<a href=\"/otp/send?otp_delivery_selection_form%5Botp_method%5D=voice\">" \
             "#{t('links.two_factor_authentication.voice')}</a>"
  expect(output).to eq t('instructions.2fa.sms.fallback_html', link: link)
end

it 'returns a link for sms delivery if user initially selected voice' do
  presenter = TwoFactorAuthCode::PhoneDeliveryPresenter.new(
    attributes_for(:generic_otp_presenter).merge!(delivery_method: 'voice')
  )
  output = presenter.phone_fallback_link

  link = "<a href=\"/otp/send?otp_delivery_selection_form%5Botp_method%5D=sms\">" \
             "#{t('links.two_factor_authentication.sms')}</a>"
  expect(output).to eq t('instructions.2fa.voice.fallback_html', link: link)
end
```